holopy/scattering/theory/scatteringtheory.py

```diff
@@ -86,76 +79,17 @@ def _calculate_scattered_field(self, scatterer, schema):
-                this_field = self._calculate_scattered_field(
-                    scatterer.select({illumination: illum}), this_schema)
-                field.append(this_field)
-            field = clean_concat(field, dim=schema.illum_wavelen.illumination)
-        else:
-            # See if we can handle the scatterer in one step
-            if self._can_handle(scatterer):
-                field = self._get_field_from(scatterer, schema)
-            # FIXME this checks if it is a composite, but does not check
-            # if each element of the composite can be handled by the theory.
-            elif isinstance(scatterer, Scatterers):
-                # if it is a composite, try superposition
-                scatterers = scatterer.get_component_list()
-                field = self._get_field_from(scatterers[0], schema)
-                for s in scatterers[1:]:
-                    field += self._get_field_from(s, schema)
-            else:
-                raise TheoryNotCompatibleError(self, scatterer)
-
-        return field
-
-    def _get_field_from(self, scatterer, schema):
-        # FIXME this checks a global value in a loop.
-        wavevector = get_wavevec_from(schema)
-        if isinstance(scatterer, Sphere) and scatterer.center is None:
+        if scatterer.center is None:
             raise MissingParameter("center")
-        positions = sphere_coords(
-            schema, scatterer.center, wavevec=wavevector)  # 8.6 ms !!
-        field = np.transpose(
-            self._raw_fields(
-                stack_spherical(positions),
-                scatterer,
-                medium_wavevec=wavevector,
-                medium_index=schema.medium_index,
-                illum_polarization=schema.illum_polarization)
-            )
-        phase = np.exp(-1j * wavevector * scatterer.center[2])
-        # TODO: fix and re-enable internal fields
-        # if self._scatterer_overlaps_schema(scatterer, schema):
-        #     inner = scatterer.contains(schema.positions.xyz())
-        #     field[inner] = np.vstack(
-        #         self._raw_internal_fields(positions[inner].T, s,
-        #                                  optics)).T
-        field *= phase
-        dimstr = primdim(positions)
-
-        # FIXME why is this here? Since ``positions = sphere_coords(...)``
-        # shouldn't ``positions`` always be an xr.DataArray?
-        if isinstance(positions[dimstr], xr.DataArray):
-            coords = {key: (dimstr, val.values)
-                      for key, val in positions[dimstr].coords.items()}
-        else:
-            coords = {key: (dimstr, val) for key, val in positions.items()}
-        coords = updated(coords, {dimstr: positions[dimstr],
-                                  vector: ['x', 'y', 'z']})
-        field = xr.DataArray(field, dims=[dimstr, vector], coords=coords,
-                             attrs=schema.attrs)
+        is_multicolor_hologram = len(ensure_array(schema.illum_wavelen)) > 1
```

Comment:
you should really be checking if the schema has a dimension called 'illumination'. My intention was that it could refer to different colours and/or different polarizations (with possible extensions to multiple directions, etc. in future), not just illum_wavelen.

Reply:
I confirmed that this is what it checked before - I don't think it's the best way to do it but that's not your fault. Since there's no change in behaviour I see no problem here.

Reply:
Marking as resolved :)